src/gui/editors/PianoRoll.cpp

```diff
@@ -4137,8 +4137,15 @@ void PianoRoll::finishRecordNote(const Note & n )
 					Note n1( n.length(), it->pos(),
 							it->key(), it->getVolume(),
 							it->getPanning() );
-					n1.quantizeLength( quantization() );
-					m_midiClip->addNote( n1 );
+
+					// TODO this value is probably better off cached somewhere;
+					//  currently doing a string parse everytime a note is released.
+					bool doQuantize = (ConfigManager::inst()->value("midi", "autoquantize").toInt() != 0);
```

Comment:
Probably should make this a member of ``PianoRoll`` and make sure it gets updated with ``ConfigManager::valueChanged`` somewhere.
PS: You don't need the parentheses if you could plan on keeping this code.

Reply:
Implemented and ready for testing.